packages/gatsby-theme-project-portal/gatsby-node.js

```diff
@@ -129,3 +194,157 @@ exports.createPages = async ({ graphql, actions, reporter }) => {
+    let documents = []
+    Object.keys(searchNodes).forEach((page) => {
+      if (page === "siteUrl") {
+        return
+      } else {
+        searchNodes[page].nodes.forEach((node) => {
+          let tempNode = structuredClone(node)
+          if (page === "allProject") {
+            if (!node.slug.includes("/")) {
+              tempNode.slug = `project/${node.slug}`
+            }
+          }
+          const newItem = Object.values(tempNode).map((field) => {
+            if (lodash.isNull(field)) {
+              return null
+            }
+
+            if (typeof field === "object") {
+              return Object.values(field)
+                .filter((value) => !lodash.isNull(value))
+                .map((value) =>
+                  typeof value === "string"
+                    ? value
+                    : Object.values(value)
+                        .filter((k) => !lodash.isNull(k))
+                        .join(" ")
+                )
+                .join(" ")
+            } else {
+              return field
+            }
+          })
+
+          Object.keys(tempNode).forEach((page, i) => {
+            tempNode[page] = newItem[i]
+          })
+          documents.push(tempNode)
+        })
+      }
+    })
+
+    let index = lunr(function () {
+      this.ref("slug") // identifier
+      // fields to search
+      this.field("slug")
+      this.field("faq")
+      this.field("aims")
+      this.field("lede")
+      this.field("title")
+      this.field("question")
+      this.field("status")
+      this.field("agency")
+      this.field("topics")
+      this.field("mainContact")
+      this.field("summary")
+      this.field("status")
+      this.field("purpose")
+      this.field("emailContent")
+      this.field("projectTeam")
+      this.field("expertise")
+      this.field("requirement")
+
+      // allow access to the position of the found word in the field
+      this.metadataWhitelist = ["position"]
+      /* allow stop words (see https://github.com/olivernn/lunr.js/blob/master/lunr.js)
+             we have 'about' as one of the slugs, which spurred this decision
+             We can implement our own stopWordFilter:
+                      var idx = lunr(function () {
+                        // normal index definition
+                        this.pipeline.after(lunr.stopWordFilter, myStopWordFilter)
+                         this.pipeline.remove(lunr.stopWordFilter)
+                      })
+          */
+      this.pipeline.remove(lunr.stopWordFilter)
+
+      documents.forEach(function (doc) {
+        this.add(doc)
+      }, this)
+    })
+    return { index, documents }
+  }
+
+  const { index, documents } = createSearchIndex({ allProject, allGeneralPage })
+  fs.writeFile("static/lunr-index.json", JSON.stringify(index), (err) => {
+    if (err) console.error(err)
+  })
+  fs.writeFile("static/documents.json", JSON.stringify(documents), (err) => {
+    if (err) console.error(err)
+  })
```

Comment:
I'm not 100% certain but I believe the code we have currently is fine because the exports.onPostBuild section is meant for running asynchronous functions so it will complete the writeFile before serving the site (the async key word on line 198) 

Another option is for us to make the writeFile synchronous rather than asynchronous. 

```
try {
    fs.writeFileSync("static/lunr-index.json", JSON.stringify(index));
    console.log('lunr-index.json written successfully.');

   fs.writeFileSync("static/documents.json", JSON.stringify(documents));
    console.log('documents.json written successfully.');
  } catch (err) {
    console.error('Error writing files:', err);
  }
```

This should guarantee the file write is done before the build process is wrapped up. But it does essentially block the build process until the write operation is complete, which could affect performance overall

Reply:
@jashlu onPostBuild is also async for awaiting the graphql query used to create the index -- will making the writeFile async still solve our issue?

Reply:
I believe that should do it ..